src/mono/wasm/runtime/loader/exit.ts

```diff
@@ -2,9 +2,31 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 import { ENVIRONMENT_IS_NODE, ENVIRONMENT_IS_SHELL, ENVIRONMENT_IS_WEB, INTERNAL, loaderHelpers, runtimeHelpers } from "./globals";
-import { mono_log_debug, consoleWebSocket, mono_log_error, mono_log_info_no_prefix } from "./logging";
+import { mono_log_debug, consoleWebSocket, mono_log_error, mono_log_info_no_prefix, mono_log_warn } from "./logging";
+
+
+let ABORT = false;
+let EXIT = false;
+
+export function is_exited() {
+    return EXIT;
+}
+
+export function is_runtime_running() {
+    return runtimeHelpers.runtimeReady && !EXIT && !ABORT;
+}
+
+
+export function assert_runtime_running() {
+    mono_assert(runtimeHelpers.runtimeReady, "mono runtime didn't start yet");
+    mono_assert(!ABORT, "mono runtime was already aborted");
```

Comment:
we should probably use terminology like "startup of the mono runtime was aborted" if that's what ABORT means. otherwise it's hard to tell the difference between abort and exit